### PR TITLE
ListBox: Java examples

### PR DESCRIPTION
**Pull request by anezthes**
### Selection

`ListBoxCustomItemPresentation.java` and `ListBoxMultiSelection.java` use the following approach for selecting items:
```
List<Person> items = DataService.getPeople(20);
listBox.setItems(items);
listBox.select(items.get(0), items.get(3));
```
That last line seems a bit random. Is there a better way?

### Disabled items

I might have over-engineered `ListBoxDisabledItems.java`. Using constants (see `ListBoxSingleSelection.java`) didn't seem right considering the example.

### CODE REVIEW

**Review by sosa-vaadin — CHANGES_REQUESTED** [review]
- There's a repeated method in `Person.java` file.
- Some imports are missing `Frontend/` in the TS samples.
Other than these, looks good to me.

File: articles/ds/components/list-box/index.asciidoc [hunk 1]
```diff
@@ -5,6 +5,7 @@ tab-title: Usage
 page-links:
   - https://github.com/vaadin/vaadin-flow-components/releases/tag/{moduleMavenVersion:com.vaadin:vaadin-list-box-flow}[Flow {moduleMavenVersion:com.vaadin:vaadin-list-box-flow}]
   - https://github.com/vaadin/vaadin-list-box/releases/tag/v{moduleNpmVersion:vaadin-list-box}[Web Component {moduleNpmVersion:vaadin-list-box}]
+section-nav: incomplete
```
Comment (jouni) [inline articles/ds/components/list-box/index.asciidoc:8]:
```suggestion
```

File: articles/ds/components/list-box/index.asciidoc [hunk 2]
```diff
@@ -16,8 +17,8 @@ List Box allows the user to select one or more values from a scrollable list of
 .Work In Progress
 [NOTE]
 ====
-The Java examples for this component are currently in progress.
-In the meantime, you can view the previous examples.
+The usage examples for this component are currently in progress.
+In the meantime, you can view the previous examples for the Java/Flow component.
```
Comment (jouni) [inline articles/ds/components/list-box/index.asciidoc:21]: Remove this admonition block, as both TS and Java examples are now available.

File: frontend/demo/component/listbox/list-box-custom-item-presentation.ts [hunk 3]
```diff
@@ -28,25 +29,27 @@ export class Example extends LitElement {
   render() {
     return html`
       <!-- tag::snippet[] -->
-      <vaadin-list-box multiple .selectedValues="${[0, 2]}">
+      <vaadin-list-box multiple .selectedValues=${[0, 2]}>
```
Comment (jouni) [inline frontend/demo/component/listbox/list-box-custom-item-presentation.ts:32]:
```suggestion
      <vaadin-list-box multiple .selectedValues="${[0, 2]}">
```

File: frontend/demo/component/listbox/list-box-custom-item-presentation.ts [hunk 4]
```diff
@@ -28,25 +29,27 @@ export class Example extends LitElement {
...
-                <div>
-                  ${person.firstName} ${person.lastName}
-                  <div
-                    style="font-size: var(--lumo-font-size-s); color: var(--lumo-secondary-text-color);"
+          person => html`
+            <vaadin-item style="line-height: var(--lumo-line-height-m);">
+              <vaadin-horizontal-layout style="align-items: center;" theme="spacing">
+                <vaadin-avatar
+                  .img=${person.pictureUrl}
+                  .name=${`${person.firstName} ${person.lastName}`}
```
Comment (jouni) [inline frontend/demo/component/listbox/list-box-custom-item-presentation.ts:39]:
```suggestion
                  .img="${person.pictureUrl}"
                  .name="${`${person.firstName} ${person.lastName}`}"
```

File: frontend/demo/component/listbox/list-box-multi-selection.ts [hunk 5]
```diff
@@ -15,23 +14,25 @@ export class Example extends LitElement {
...
-        <vaadin-item>Aria Bailey</vaadin-item>
-        <vaadin-item>Aaliyah Butler</vaadin-item>
-        <vaadin-item>Eleanor Price</vaadin-item>
-        <vaadin-item>Allison Torres</vaadin-item>
-        <vaadin-item>Madeline Lewis</vaadin-item>
-        <vaadin-item>Lucas Edwards</vaadin-item>
-        <vaadin-item>Paul Gibbs</vaadin-item>
-        <vaadin-item>Hana Newton</vaadin-item>
-        <vaadin-item>Jane Stone</vaadin-item>
+      <vaadin-list-box multiple .selectedValues=${[0, 3]} style="height: 200px">
```
Comment (jouni) [inline frontend/demo/component/listbox/list-box-multi-selection.ts:28]:
```suggestion
      <vaadin-list-box multiple .selectedValues="${[0, 3]}" style="height: 200px">
```

File: src/main/java/com/vaadin/demo/domain/Person.java [hunk 6]
```diff
@@ -32,6 +32,8 @@
 
   private boolean manager;
 
+  public String getFullName() { return firstName + " " + lastName; }
+
```
[the remark this answers is not in the document]
Comment (anezthes) [reply src/main/java/com/vaadin/demo/domain/Person.java]: Thanks, fixed!

File: frontend/demo/component/listbox/list-box-multi-selection.ts (hunk 5 above)
Comment (anezthes) [reply frontend/demo/component/listbox/list-box-multi-selection.ts:28]: Fixed!

File: frontend/demo/component/listbox/list-box-custom-item-presentation.ts (hunk 4 above)
Comment (anezthes) [reply frontend/demo/component/listbox/list-box-custom-item-presentation.ts:39]: Fixed!

File: frontend/demo/component/listbox/list-box-custom-item-presentation.ts (hunk 3 above)
Comment (anezthes) [reply frontend/demo/component/listbox/list-box-custom-item-presentation.ts:32]: Fixed!

File: articles/ds/components/list-box/index.asciidoc (hunk 2 above)
Comment (anezthes) [reply articles/ds/components/list-box/index.asciidoc:21]: Removed!

File: articles/ds/components/list-box/index.asciidoc (hunk 1 above)
Comment (anezthes) [reply articles/ds/components/list-box/index.asciidoc:8]: Removed!

File: frontend/demo/component/listbox/list-box-single-selection.ts [hunk 7]
```diff
@@ -1,11 +1,10 @@
...
 import { html, LitElement, customElement } from 'lit-element';
-import '@vaadin/vaadin-list-box';
-import '@vaadin/vaadin-item/vaadin-item';
-import '@vaadin/vaadin-icons/vaadin-icons';
 import '@polymer/iron-icon';
-import { applyTheme } from 'Frontend/generated/theme';
+import '@vaadin/vaadin-icons/vaadin-icons';
+import '@vaadin/vaadin-item/vaadin-item';
+import '@vaadin/vaadin-list-box';
+import { applyTheme } from 'generated/theme';
```
[the remark this answers is not in the document]
Comment (anezthes) [reply frontend/demo/component/listbox/list-box-single-selection.ts:7]: Fixed!

File: frontend/demo/component/listbox/list-box-separators.ts [hunk 8]
```diff
@@ -1,11 +1,8 @@
...
-
+import '../../init'; // hidden-full-source-line
 import { html, LitElement, customElement } from 'lit-element';
-import '@vaadin/vaadin-list-box';
 import '@vaadin/vaadin-item/vaadin-item';
-import '@vaadin/vaadin-icons/vaadin-icons';
-import '@polymer/iron-icon';
-import { applyTheme } from 'Frontend/generated/theme';
+import '@vaadin/vaadin-list-box';
+import { applyTheme } from 'generated/theme';
```
[the remark this answers is not in the document]
Comment (anezthes) [reply frontend/demo/component/listbox/list-box-separators.ts:5]: Fixed!

File: frontend/demo/component/listbox/list-box-multi-selection.ts [hunk 9]
```diff
@@ -1,11 +1,10 @@
...
-import { html, LitElement, customElement } from 'lit-element';
-import '@vaadin/vaadin-list-box';
+import '../../init'; // hidden-full-source-line
+import { html, LitElement, customElement, internalProperty } from 'lit-element';
 import '@vaadin/vaadin-item/vaadin-item';
-import '@vaadin/vaadin-icons/vaadin-icons';
-import '@polymer/iron-icon';
-import { applyTheme } from 'Frontend/generated/theme';
+import '@vaadin/vaadin-list-box';
+import { applyTheme } from 'generated/theme';
```
[the remark this answers is not in the document]
Comment (anezthes) [reply frontend/demo/component/listbox/list-box-multi-selection.ts:5]: Fixed!

File: frontend/demo/component/listbox/list-box-disabled-items.ts [hunk 10]
```diff
@@ -1,11 +1,8 @@
...
-
+import '../../init'; // hidden-full-source-line
 import { html, LitElement, customElement } from 'lit-element';
-import '@vaadin/vaadin-list-box';
 import '@vaadin/vaadin-item/vaadin-item';
-import '@vaadin/vaadin-icons/vaadin-icons';
-import '@polymer/iron-icon';
-import { applyTheme } from 'Frontend/generated/theme';
+import '@vaadin/vaadin-list-box';
+import { applyTheme } from 'generated/theme';
```
[the remark this answers is not in the document]
Comment (anezthes) [reply frontend/demo/component/listbox/list-box-disabled-items.ts:5]: Fixed!

File: frontend/demo/component/listbox/list-box-custom-item-presentation.ts [hunk 11]
```diff
@@ -1,13 +1,14 @@
...
-import '@vaadin/vaadin-icons/vaadin-icons';
-import '@polymer/iron-icon';
-import Person from 'Frontend/generated/com/vaadin/demo/domain/Person';
-import { getPeople } from 'Frontend/demo/domain/DataService';
-import { applyTheme } from 'Frontend/generated/theme';
+import '@vaadin/vaadin-list-box';
+import '@vaadin/vaadin-lumo-styles/typography';
+import '@vaadin/vaadin-ordered-layout/vaadin-horizontal-layout';
+import '@vaadin/vaadin-ordered-layout/vaadin-vertical-layout';
+import { applyTheme } from 'generated/theme';
```
[the remark this answers is not in the document]
Comment (anezthes) [reply frontend/demo/component/listbox/list-box-custom-item-presentation.ts:9]: Fixed!

File: frontend/demo/component/listbox/list-box-basic.ts [hunk 12]
```diff
@@ -1,11 +1,8 @@
-import 'Frontend/demo/init'; // hidden-full-source-line
-
+import '../../init'; // hidden-full-source-line
 import { html, LitElement, customElement } from 'lit-element';
 import '@vaadin/vaadin-list-box/vaadin-list-box';
 import '@vaadin/vaadin-item/vaadin-item';
-import '@vaadin/vaadin-icons/vaadin-icons';
-import '@polymer/iron-icon';
-import { applyTheme } from 'Frontend/generated/theme';
+import { applyTheme } from 'generated/theme';
```
[the remark this answers is not in the document]
Comment (anezthes) [reply frontend/demo/component/listbox/list-box-basic.ts:5]: Fixed!

**Review by sosa-vaadin — APPROVED** [review]
Fix conflicts with `latest` branch and we're good to go.

**Pull request merged**